### PR TITLE
add resources

### PR DESCRIPTION
**Pull request by lumisphere902**
Handle resources when adding/fetching events.

One weird thing: all the current databases have both an auto inc ID as the primary key, as well as a UUID. It seems to me like the UUID should be the primary key, as I don't think the auto inc ID is used anywhere. However, I followed this pattern just in case I'm missing something.

### CODE REVIEW

File: app/db/schema/resource.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+      },
+    },
+
+    // resource value
+    value: {
+      type: Sequelize.STRING,
+      allowNull: false,
+      validate: {
+        len: {
+          args: [3, 255],
```
Comment (mattxwang) [inline app/db/schema/resource.js:60]: Can we make this max length longer? See https://github.com/uclaacm/membership-portal-ui/issues/81

File: app/util/resource_utils.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
+const { Resource } = require('../db');
+
+const populateEventsWithResources = (events) => {
```
Comment (mattxwang) [inline app/util/resource_utils.js:3]: Is there a way we'd have this call `populateOneEventWithResources`? seems like a reasonable-ish way to de-duplicate logic

File: app/db/schema/resource.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+          args: 4,
+          msg: 'Invalid value for event UUID',
+        },
+        notEmpty: {
+          msg: 'The event UUID is a required field',
+        },
+      },
+    },
+
+    // resource type
```
Comment (mattxwang) [inline app/db/schema/resource.js:32]: i'm a bit out of the loop, is the plan for `type` to be a free-floating string, or for it to be an enum?

(if it's the former, we probably want to enforce a length min/max)

File: app/db/schema/resource.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+      type: Sequelize.STRING,
+      validate: {
+        len: {
+          args: [3, 255],
+          msg: 'The title must be between 3 and 255 characters long',
+        },
+      },
+    },
+
+    // resource value
```
Comment (mattxwang) [inline app/db/schema/resource.js:54]: Will resource ever not be a link? If so, we may want to validate the link-ness!

**Review by mattxwang — CHANGES_REQUESTED** [review]
Very cursory review, but we can talk about it more @ dev team meeting!

**Comment by mattxwang** [comment]
> One weird thing: all the current databases have both an auto inc ID as the primary key, as well as a UUID. It seems to me like the UUID should be the primary key, as I don't think the auto inc ID is used anywhere. However, I followed this pattern just in case I'm missing something.

Extremely strange, though also I imagine reformatting might be a bit of a nightmare. Let's follow up with this in another issue, warrants investigation.

File: app/db/schema/resource.js (hunk 1 above)
Comment (lumisphere902) [reply app/db/schema/resource.js:60]: OK, it's just 255 because the String type is varchar(255). What's a reasonable max length? 1000?

**Comment by lumisphere902** [comment]
Closing due to inactivity

**Pull request closed without merging**